### PR TITLE
[WIP][MVT] cleaned up mvt branch

### LINKED ISSUE #3066: create a new clean branch based on mvt branch for support of /gl-styles
the old mvt branch is not up to date with master, webmercator is already merged.

please make a PR that uses the refactoring done in [mvt branch](https://github.com/geoadmin/mf-chsdi3/pull/3005), and test.

After this is done, use this branch in geoadmin3:mvt branch (PR)

### PR DESCRIPTION
**Pull request by boecklic**
this PR replaces PR #3005 and resolves #3066 
branch `mvt_clean` is based on recent master (70efc81e149c3c343cd42daa572d3e20a03ca434) with cherry-picking commits in #3005. If new upstream changes from master should be incorporated in this branch, it should be `rebased` on `master` instead of merging upstream `master` into `mvt_clean`.

### CODE REVIEW

File: chsdi/models/vector/__init__.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def __read__(self):
     def srid(self):
         return self.geometry_column().type.srid
 
-    def to_esrijson(self, trans, returnGeometry):
+    def to_esrijson(self, trans, returnGeometry, out_srid):
```
Comment (pakb) [inline chsdi/models/vector/__init__.py]: do we need this param? it doesn't seem to be used anywhere

File: chsdi/models/vector/__init__.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def to_esrijson(self, trans, returnGeometry):
                                    layerName=trans(self.__bodId__))
         return self._no_geom_template(trans)
 
-    def to_geojson(self, trans, returnGeometry):
+    def to_geojson(self, trans, returnGeometry, out_srid):
```
Comment (pakb) [inline chsdi/models/vector/__init__.py]: Same as above, new param, but no usage in the definition

File: chsdi/models/vector/__init__.py (hunk 1 above)
Comment (boecklic) [reply chsdi/models/vector/__init__.py]: We already discussed this during code review. Parameter is not used here, but calling function passes three args (https://github.com/geoadmin/mf-chsdi3/blob/mvt_clean/chsdi/views/features.py#L760), if removed, all calling functions should be changed as well. @procrastinatio needed or not? you set the attribute `wkid=self.geometry_column().type.srid_out` when constructing the Feature, was the intention to use `out_srid` here?

**Comment by danduk82** [comment]
@boecklic , the first 4 commits of this branch are already in master, and there are a few changes. Please create a branch based on `master` and cherry-pick only the commits done by @loicgasser (and then fix the possible issues). 
You might as well remove `out_srid` from the code after that.

**Comment by boecklic** [comment]
Marc just told me. I quickly crosschecked and not all changes concerning webmerc from this branch are in master as well (ex. https://github.com/geoadmin/mf-chsdi3/pull/3005/commits/dac9440f0558afebc60c1bab83bd47758c69c9f0#diff-dc7d9fa658c78f47efa836fded30e6ad) Can I assume that the necessary stuff for basic webmerc support is in master and that we can discard what's left of it here?

**Comment by danduk82** [comment]
mmh, @procrastinatio it seems to me that all is in master, can you check that please?. If something for webmerc is missing, yes please add it.

**Pull request closed without merging**